tests/test_asgi.py

```diff
@@ -0,0 +1,262 @@
+from __future__ import annotations
+
+import asyncio
+import io
+import os
+import stat
+import tempfile
+
+import pytest
+
+from whitenoise.asgi import (
+    AsgiWhiteNoise,
+    convert_asgi_headers,
+    convert_wsgi_headers,
+    read_file,
+    receive_request,
+    serve_static_file,
+)
+from whitenoise.responders import StaticFile
+
+from .test_whitenoise import application as whitenoise_application
+from .test_whitenoise import files
+
+
+@pytest.fixture()
+def loop():
+    return asyncio.get_event_loop()
+
+
+class MockStat:
```

Comment:
could maybe use `SimpleNamespace` in the one use site rather than create a class

https://adamj.eu/tech/2022/01/14/make-simple-mocks-with-simplenamespace/